client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningGetResultsIT.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.client;
+
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.protocol.xpack.ml.GetBucketsRequest;
+import org.elasticsearch.protocol.xpack.ml.GetBucketsResponse;
+import org.elasticsearch.protocol.xpack.ml.PutJobRequest;
+import org.elasticsearch.protocol.xpack.ml.job.config.Job;
+import org.elasticsearch.protocol.xpack.ml.job.results.Bucket;
+import org.elasticsearch.protocol.xpack.ml.job.util.PageParams;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+public class MachineLearningGetResultsIT extends ESRestHighLevelClientTestCase {
+
+    private static final String RESULTS_INDEX = ".ml-anomalies-shared";
+    private static final String DOC = "doc";
+
+    private static final String JOB_ID = "get-results-it-job";
+
+    private BucketStats bucketStats = new BucketStats();
+
+    @Before
+    public void createJobAndIndexResults() throws IOException {
+        MachineLearningClient machineLearningClient = highLevelClient().machineLearning();
+        Job job = MachineLearningIT.buildJob(JOB_ID);
+        machineLearningClient.putJob(new PutJobRequest(job), RequestOptions.DEFAULT);
+
+        BulkRequest bulkRequest = new BulkRequest();
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+        long time = 1533081600000L; // 2018-08-01T00:00:00Z
+        long endTime = time + 3600000L * 24 * 10; // 10 days of hourly buckets
+        while (time < endTime) {
+            addBucketIndexRequest(time, false, bulkRequest);
+            addRecordIndexRequests(time, false, bulkRequest);
+            time += 3600000L;
+        }
+
+        // Also index an interim bucket
+        addBucketIndexRequest(time, true, bulkRequest);
+        addRecordIndexRequests(time, true, bulkRequest);
+
+        highLevelClient().bulk(bulkRequest, RequestOptions.DEFAULT);
+    }
+
+    private void addBucketIndexRequest(long timestamp, boolean isInterim, BulkRequest bulkRequest) {
+        IndexRequest indexRequest = new IndexRequest(RESULTS_INDEX, DOC);
+        double bucketScore = randomDoubleBetween(0.0, 100.0, true);
+        bucketStats.report(bucketScore);
+        indexRequest.source("{\"job_id\":\"" + JOB_ID + "\", \"result_type\":\"bucket\", \"timestamp\": " + timestamp + "," +
```

Comment:
Any reason why you prefer to build the string representation rather than constructing a Bucket object converting it to a ByteReference and use the overload of `indexRequest.source`?

Reply:
The constructor and setter methods of `Bucket` are package protected. I like that as it means the users cannot modify a bucket.